### PR TITLE
Feat: Claim streaming payments

### PR DESCRIPTION
**Pull request by davecreaser**
## Description

- Create a saga to claim a streaming payment payout.
- Store amounts claimed for streaming payments in the db.
- Create hook for calculating the amounts currently available to claim, and the amounts claimed up to now.

Block Ingestor counterpart here: https://github.com/JoinColony/block-ingestor/pull/219

## Testing

Ensure that you're running the block ingestor from the above PR.

Test that you can claim streaming payments successfully:

* Step 1. Turn on streaming payments extension.
* Step 2. Go to the dashboard and look at the temporary UI. Create a streaming payment using the 'Create Streaming Payment' button after typing an amount into the input.
* Step 3. Scroll down a little, and type in the native ID of the streaming payment (eg: '1'). It should be found, and show you the amount available to claim. By default in the testing UI these streaming payments are created with an interval of 1 day, a start time of 1 week ago, and an end time of 1 week in the future.
* Step 4. Click the 'Claim' button in the streaming payments section of the temp UI.
* Step 5. Check that the transaction passed. Refresh the page and see if your balance went up (you should be the recipient by default). Find the streaming payment again using the UI. You should see the amount you claimed, and the next available amount slowly going up every time you refresh.

Screenshot 2024-05-07 at 16 18 32

## Diffs

**New stuff** ✨

* New saga for claiming streaming payments.
* New hook for calculating the amounts currently available to claim, and the amounts claimed up to now for a streaming payment.

**Changes** 🏗

* Changed the schema for streaming payments in the db to add 'claims'.

Contributes to #2206

### CODE REVIEW

**Review by ArmandoGraterol — COMMENTED** [review]
looking good to me! Just 1 question... I've followed the testing instructions and seems like I was able to claim successfully and see the change reflected on my balance, however, the `Amount claimed to date` is still `0`. Is this expected?
Screenshot 2024-05-07 at 14 10 12

I would expect this amount to change to the 7k I just claimed

**Comment by davecreaser** [comment]
> looking good to me! Just 1 question... I've followed the testing instructions and seems like I was able to claim successfully and see the change reflected on my balance, however, the `Amount claimed to date` is still `0`. Is this expected? Screenshot 2024-05-07 at 14 10 12
> 
> I would expect this amount to change to the 7k I just claimed

Hmm, not expected! Is it still 0 after refreshing the page? (I haven't put in any auto updating of that number since it's just a temporary UI, but after a refresh it should be updated)

**Comment by davecreaser** [comment]
Okay cool cool, definitely a bug then! I'll look into it, thanks :)

File: src/hooks/useStreamingPaymentAmountsLeft.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          return { amounts, [payout.tokenAddress]: newAmount };
+        }
+
+        const amountAvailableSinceStart = BigNumber.from(payout.amount)
+          .mul(
+            BigNumber.from(durationToClaim)
+              .mul(BigNumber.from(10).pow(18))
+              .div(BigNumber.from(streamingPayment.interval)),
+          )
+          .div(BigNumber.from(10).pow(18));
```
Comment (jakubcolony) [inline src/hooks/useStreamingPaymentAmountsLeft.ts:51]: I was going to make a comment about using hardcoded 18 as token decimals but upon further reading further on this I suppose it's a good representation of how solidity does multiplication - https://github.com/dapphub/ds-math?tab=readme-ov-file#wmul. Nice work figuring this out 👍

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Gotta say I love the new UI! 💯 

I left a few comments but it's a very solid piece of work, especially considering it's your first "proper" issue - so well done @davecreaser.

Functionally everything seems to be on point, but after pulling latest changes I still can't get the "claimed to date" amount to work:
image

It seems like no claims are being created in the DB:
image

Happy to jump on a call tomorrow to debug (unless of course I'm doing something wrong?)

**Comment by davecreaser** [comment]
@jakubcolony and @ArmandoGraterol could you please try again and see if the amount claimed to date is now correct? I realised that I never updated the block ingestor hash in the docker files so unless you ran the block ingestor separately then it wouldn't have worked :) It is updated now though, so npm run dev should be enough!

File: src/hooks/useStreamingPayments.ts
[the remark this answers is not in the document]
Comment (davecreaser) [reply src/hooks/useStreamingPayments.ts:1]: Oh yeah, hadn't spotted that. Added it now!

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Thanks for the changes @davecreaser, just one tiny detail before you merge - we seem to be missing i18n messages for claim transaction - could you please have a look at the `en.json` file? While you're there, it'd be great if you could also add one for create stream transaction, I must have missed it the first time!

image

**Review by iamsamgibbs — APPROVED** [review]
Nice job on this one, everything is working great.

Screenshot 2024-05-09 at 11 15 55

Balance is updating correctly.

Screenshot 2024-05-09 at 11 16 17

il8n messages look good.

Screenshot 2024-05-09 at 11 16 24

And available amount to claim is working as expected.

Screenshot 2024-05-09 at 11 16 39

Nice job!

**Review by jakubcolony — APPROVED** [review]
LGTM, great work 👍

**Review by iamsamgibbs — APPROVED** [review]
All looks good! Thanks for dealing with the rebase!

**Pull request merged**